lib/react-dom.js

```diff
@@ -404,19 +404,281 @@ declare type $JSXIntrinsics = {
   tspan: ReactDOM$SVGElementJSXIntrinsic,
   use: ReactDOM$SVGElementJSXIntrinsic,
   // Elements React adds extra props for.
-  input: {instance: HTMLInputElement, props: {children?: React$Node, [key: string]: any}},
-  textarea: {instance: HTMLTextAreaElement, props: {children?: React$Node, [key: string]: any}},
-  select: {instance: HTMLSelectElement, props: {children?: React$Node, [key: string]: any}},
+  input: ReactDOM$HTMLElementJSXIntrinsic<HTMLInputElement, ReactDOM$VoidElementProps<ReactDOM$HTMLElementProps>>,
+  textarea: ReactDOM$HTMLElementJSXIntrinsic<HTMLTextAreaElement, ReactDOM$HTMLElementProps>,
+  select: ReactDOM$HTMLElementJSXIntrinsic<HTMLSelectElement, ReactDOM$HTMLElementProps>,
   // Catch-all for custom elements.
-  [string]: ReactDOM$HTMLElementJSXIntrinsic,
+  [string]: ReactDOM$HTMLElementJSXIntrinsic<>,
 };
 
-type ReactDOM$HTMLElementJSXIntrinsic = {
-  instance: HTMLElement,
-  props: {children?: React$Node, [key: string]: any},
+type ReactDOM$GlobalEventHandlers<E> = {
+  onAbort?: (SyntheticEvent<E>) => void,
+  onAbortCapture?: (SyntheticEvent<E>) => void,
+  onAnimationEnd?: (SyntheticAnimationEvent<E>) => void,
+  onAnimationEndCapture?: (SyntheticAnimationEvent<E>) => void,
+  onAnimationIteration?: (SyntheticAnimationEvent<E>) => void,
+  onAnimationIterationCapture?: (SyntheticAnimationEvent<E>) => void,
+  onAnimationStart?: (SyntheticAnimationEvent<E>) => void,
+  onAnimationStartCapture?: (SyntheticAnimationEvent<E>) => void,
+  onAuxClick?: (SyntheticMouseEvent<E>) => void,
+  onAuxClickCapture?: (SyntheticMouseEvent<E>) => void,
+  onBeforeInput?: (SyntheticInputEvent<E>) => void,
+  onBeforeInputCapture?: (SyntheticInputEvent<E>) => void,
+  onBlur?: (SyntheticFocusEvent<E>) => void,
+  onBlurCapture?: (SyntheticFocusEvent<E>) => void,
+  onCancel?: (SyntheticEvent<E>) => void,
+  onCancelCapture?: (SyntheticEvent<E>) => void,
+  onCanPlay?: (SyntheticEvent<E>) => void,
+  onCanPlayCapture?: (SyntheticEvent<E>) => void,
+  onCanPlayThrough?: (SyntheticEvent<E>) => void,
+  onCanPlayThroughCapture?: (SyntheticEvent<E>) => void,
+  onChange?: (SyntheticEvent<E>) => void,
```

Comment:
`SyntheticEvent` has a `target` property of type `EventTarget` which is missing `value`. This, however, seems to be common in examples. See https://reactjs.org/docs/forms.html#handling-multiple-inputs

Reply:
Same holds for properties like `checked`, and less frequently for `files`, `id`, `blur, etc.

Reply:
This probably has to do with the typing of `target` in `SyntheticEvent`:

https://github.com/facebook/flow/pull/7569/files#diff-ce626ccb6b8eb922c17388db8d33cbabR137

Reply:
`target` is fine as is, `currentTarget` should be used

Reply:
> `currentTarget` should be used

Not on input or select elements, where we most likely don't need/want event propagation.

Reply:
I usually type `onChange` handlers as `onChange?: (SyntheticInputEvent<>) => void`.   
Does it means it will break when this lands?

Reply:
Do we have some sort of consensus here? :D

Reply:
@pascaldues, what you mean by event propagation? it IS `target` that propagates events

Reply:
@panagosg7 @pascalduez @goodmind so what should this type actually be?

Reply:
@panagosg7, @goodmind pointed to this in Discord: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/11508#issuecomment-256045682

Does this make sense to you?